services/src/main/scala/cromwell/services/MetadataServiceActor.scala

```diff
@@ -34,3 +37,22 @@ object MetadataServiceActor {
   }
   */
 }
+
+object MetadataServiceActorImplicits {
+  implicit class EnhancedServiceRegistryActorForMetadata(val actor: ActorRef) extends AnyVal {
+    def pushWdlValueMetadata(metadataKey: MetadataKey, output: WdlValue): Unit = output match {
+      case WdlArray(_, valueSeq) =>
+        val zippedSeq = valueSeq.zipWithIndex
+        zippedSeq foreach { case (value, index) => actor.pushWdlValueMetadata(metadataKey.copy(key = s"${metadataKey.key}[$index]"), value) }
+      case WdlMap(_, valueMap) =>
+        valueMap foreach { case (key, value) => actor.pushWdlValueMetadata(metadataKey.copy(key = metadataKey.key + s":$key"), value) }
+      case value =>
+        actor ! PutMetadataAction(MetadataEvent(metadataKey, MetadataValue(value)))
```

Comment:
Hmm, maybe I'll just merge with the `Any` version. I'm not sure how the precedence of `apply`s works